src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/jpa/repository/ReferralRepositoryTest.kt

```diff
@@ -412,23 +412,19 @@ class ReferralRepositoryTest @Autowired constructor(
     numberOfAssignedUsers: Int = 1,
     intervention: Intervention? = null
   ): Referral {
-
-    val assignedUsers = mutableListOf<AuthUser>().apply {
-      repeat(numberOfAssignedUsers) { this.add(element = authUserFactory.create(random(6), random(5), random(12))) }
-    }
-    val referral: Referral = intervention?.let { intervention ->
-      referralFactory.createSent(
-        intervention = intervention,
-        assignments = assignedUsers.map { ReferralAssignment(OffsetDateTime.now(), assignedBy = it, assignedTo = it) }
-      )
-    } ?: run {
-      referralFactory.createSent(
-        intervention = createIntervention(asPrime),
-        assignments = assignedUsers.map { ReferralAssignment(OffsetDateTime.now(), assignedBy = it, assignedTo = it) }
-      )
+    val assignedUsers = mutableListOf<ReferralAssignment>().apply {
+      repeat(numberOfAssignedUsers) {
+        val user = authUserFactory.create(random(6), random(5), random(12))
+        this.add(ReferralAssignment(OffsetDateTime.now(), assignedBy = user, assignedTo = user, superseded = true))
```

Comment:
Would be much better if we could use `repeat(...) { assignmentService.assign(referral, user, user) }` or similar